btx/processing/rawimagetimetool.py

```diff
@@ -0,0 +1,575 @@
+    --------
+    __init__(self, expmt: str, savedir: str) - Instantiate an analysis object for experiment expmt.
+    open_run(self, run: str) - Open DataSource for a specified run.
+    get_tt_name(self) - Determine time tool camera name in data files.
+    calibrate(self, run: str, order: int, figs=True) - Calibrate the analysis object for jitter correction on specified run.
+    process_run(self) - Perform edge detection on all events in a run.
+    detect_edge(self, img: np.array, kernel: np.array) - Perform edge detection on an image.
+    crop_image(self, img: np.array) - Crop time tool image to ROI.
+    fit_calib(self, delays: np.array, edges: np.array, amplitudes: np.array,
+            fwhm: np.array = None, order: int = 2) - Fit polynomial to calibration run data.
+    ttstage_code(self, hutch) - Return the time tool target time stage for a given hutch.(
+    edge_to_time(self, edge)
+    actual_time(self, edge, nominal)
+    timetool_correct(self, edge, nominal, model, figs=True)
+    plot_calib
+    plot_hist
+    """
+    # Class vars
+    ############################################################################
+
+    # Methods
+    ############################################################################
+
+    def __init__(self, expmt: str, savedir: str):
+        """! Initializer for the time tool analysis object.
+
+        @param expmt (str) Experiment accession name. E.g. cxilr1234.
+        @param savedir (str) Directory where any output files are saved.
+        """
+        ## @var expmt
+        # (str) Experiment accession name
+        self.expmt = expmt
+
+        ## @var hutch
+        # (str) Experimental hutch. Extracted from expmt. Needed for camera/stage accession.
+        self.hutch = self.expmt[:3]
+
+        ## @var _model
+        # (list) List of calibration model polynomial coefficients. Empty if not calibrated.
+        self._model = []
+
+        ## @var savedir
+        # (str) Directory to save output files to. Figures may be placed in a subdirectory of this one.
+        self.savedir = savedir
+
+    def open_run(self, run: str):
+        """! Open the psana DataSource and time tool detector for the specified
+        run. This method MUST be called before calibration, analysis etc. The
+        psana DataSource and Detector objects are stored in instance variables
+
+        @param run (str) Run number (e.g. 5) or range of runs (e.g. 6-19).
+        """
+
+        ## @var ds
+        # (psana.DataSource) DataSource object for accessing data for specified runs.
+        self.ds = psana.MPIDataSource(f'exp={self.expmt}:run={run}')
+
+        # Detector lookup needs to be modified for each hutch!
+        # Considering only MFX right now
+        ## @var det
+        # (psana.Detector) Detector object corresponding to the time tool camera.
+        # self.det = psana.Detector(self.get_detector_name(run), self.ds.env())
+        self.det = psana.Detector(self.get_tt_name(), self.ds.env())
```

Comment:
As discussed, I'd be in favor of using the `PsanaInterface` object here - to avoid code redundancy.